client/blocks/author-selector/index.jsx

```diff
@@ -3,84 +3,91 @@
-		onSelect: PropTypes.func,
-		exclude: PropTypes.oneOfType( [ PropTypes.arrayOf( PropTypes.number ), PropTypes.func ] ),
-		allowSingleUser: PropTypes.bool,
-		popoverPosition: PropTypes.string,
-		transformAuthor: PropTypes.func,
-	};
+	const fetchOptions = { number: 50 };
+	const trimmedSearch = search.trim?.();
 
-	static defaultProps = {
-		showAuthorMenu: false,
-		onClose: function () {},
-		allowSingleUser: false,
-		popoverPosition: 'bottom left',
-	};
-
-	state = {
-		search: '',
-	};
-
-	componentDidMount() {
-		debug( 'AuthorSelector mounted' );
+	if ( trimmedSearch ) {
+		fetchOptions.number = 20; // make search a little faster
+		fetchOptions.search = `*${ trimmedSearch }*`;
+		fetchOptions.search_columns = [ 'user_login', 'display_name' ];
 	}
 
-	render() {
-		let searchString = this.state.search || '';
-		searchString = trim( searchString );
+	const { data, isLoading, fetchNextPage, hasNextPage, isFetchingNextPage } = useUsersQuery(
+		siteId,
+		fetchOptions
+	);
 
-		const fetchOptions = {
-			siteId: this.props.siteId,
-			order: 'ASC',
-			order_by: 'display_name',
-			number: 50,
-		};
+	const users =
+		data?.users.filter( ( user ) => {
+			if ( typeof exclude === 'function' ) {
+				return ! exclude( user );
+			}
+			if ( Array.isArray( exclude ) ) {
+				return ! exclude.includes( user.ID );
+			}
+			return true;
+		} ) ?? [];
+	const listKey = [ 'authors', siteId, search ].join( '-' );
 
-		if ( searchString ) {
-			fetchOptions.number = 20; // make search a little faster
-			fetchOptions.search = searchString;
-			fetchOptions.search_columns = [ 'user_login', 'display_name' ];
-		}
+	return (
+		<SwitcherShell
+			allowSingleUser={ allowSingleUser }
+			fetchNextPage={ fetchNextPage }
+			hasNextPage={ hasNextPage }
+			ignoreContext={ ignoreContext }
+			isFetchingNextPage={ isFetchingNextPage }
+			isLoading={ isLoading }
+			listKey={ listKey }
+			onSelect={ onSelect }
+			popoverPosition={ popoverPosition }
+			search={ search }
+			siteId={ siteId }
+			transformAuthor={ transformAuthor }
+			updateSearch={ setSearch }
+			users={ users }
+		>
+			{ children }
+		</SwitcherShell>
+	);
+};
 
-		Object.freeze( fetchOptions );
-		return (
-			<SiteUsersFetcher fetchOptions={ fetchOptions } exclude={ this.props.exclude }>
-				<SwitcherShell { ...this.props } updateSearch={ this._updateSearch } />
-			</SiteUsersFetcher>
-		);
-	}
+AuthorSelector.propTypes = {
+	allowSingleUser: PropTypes.bool,
+	exclude: PropTypes.oneOfType( [ PropTypes.arrayOf( PropTypes.number ), PropTypes.func ] ),
+	onSelect: PropTypes.func,
+	popoverPosition: PropTypes.string,
+	siteId: PropTypes.number.isRequired,
+	transformAuthor: PropTypes.func,
+};
 
-	_updateSearch = ( searchTerm ) => {
-		searchTerm = searchTerm ? '*' + searchTerm + '*' : '';
-		this.setState( {
-			search: searchTerm,
-		} );
-	};
-}
+AuthorSelector.defaultProps = {
+	allowSingleUser: false,
+	popoverPosition: 'bottom left',
+	showAuthorMenu: false,
+};
```

Comment:
`defaultProps` for function components are a bit legacy. The same thing can be achieved by assigning defaults when destructuring the function parameters.

Reply:
Changed it with https://github.com/Automattic/wp-calypso/pull/51879/commits/8faf2b8b519331a2888482e20674bdcb8f33bb73 which also revealed another prop which actually wasn't used at all `showAuthorMenu` 👍🏻